src/apps/tezos/__init__.py

```diff
@@ -9,3 +9,5 @@ def boot():
     wire.add(MessageType.TezosGetAddress, __name__, "get_address", ns)
     wire.add(MessageType.TezosSignTx, __name__, "sign_tx", ns)
     wire.add(MessageType.TezosGetPublicKey, __name__, "get_public_key", ns)
+    wire.add(MessageType.TezosSignBakerOp, __name__, "sign_baker_op", ns)
```

Comment:
don't register this handler here at boot, instead register it when entering baking mode.

that way you don't need a check in `sign_baker_op`